assets/app/scripts/extensions/javaLink.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsoleExtensions.extension:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
+  .factory("ProxyPod", function(API_CFG) {
+    return function(namespace, podId, port) {
+      var kubeProxyURI = new URI(window.location.protocol + '//' + API_CFG.k8s.hostPort + API_CFG.k8s.prefix);
+      var apiVersion = API_CFG.k8s.version || 'v1beta3';
+      if (port) {
+        podId = podId + ':' + port;
+      }
+      kubeProxyURI.path(UrlHelpers.join(kubeProxyURI.path(), 'v1beta3/proxy/namespaces', namespace, 'pods', podId));
+      return kubeProxyURI.toString();
+    };
+  })
+  .run(function(ProxyPod, BaseHref, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="container.jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView(container, container.jolokiaUrl)" href="">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+      var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
```

Comment:
to avoid deep, and do checks on local objects before going scope fishing, can you reorder the guard checks and do early returns:
```
var container = $scope.container;
if (!container) {
  return
}
var jolokiaPort = _.find((container.ports || []), function(port) {
 return port.name && port.name.toLowerCase() === 'jolokia';
});
if (!jolokiaPort) {
  return;
}
var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
if (!pod) {
  return;
}
...
```